models/cartDao.js

```diff
@@ -0,0 +1,53 @@
+const { myDataSource } = require("./myDataSource");
+
+// 2. 장바구니 상품 조회
+const readCart = async (userId) => {
+  try {
+    const [cartList] = await myDataSource.query(
+      `SELECT
+        JSON_ARRAYAGG(
+          JSON_OBJECT(
+          'cartId', carts.id,
+          'thumbnail', products.thumbnail_image,
+          'productName', products.name,
+          'optionName', product_options.name, 
+          'price', products.price,
+          'quantity', carts.quantity
+          )
+        ) AS cartList
+      FROM carts
+      INNER JOIN product_options ON product_options.id = carts.product_option_id
+      INNER JOIN products ON products.id = carts.product_id
+      WHERE carts.user_id = ?;`,
+      [userId]
+    );
+    return cartList;
+  } catch (err) {
+    const error = new Error("DB_SELECT_FAILED");
+    error.statusCode = 500;
```

Comment:
에러 코드 500은 백엔드가 에러를 처리하지 못 할 때 발생하는 코드입니다.
```suggestion
    error.statusCode = 400;
```